packages/ibm-products-web-components/examples/create-side-panel/src/create-side-panel-default.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import '@carbon/ibm-products-web-components/es/components/side-panel/index.js';
+import '@carbon/web-components/es/components/text-input/index.js';
+import '@carbon/web-components/es/components/button/index.js';
+import '@carbon/web-components/es/components/number-input/index.js';
+import '@carbon/web-components/es/components/form/form-item.js';
+
+@customElement('create-side-panel-default')
+export class CreateSidepanelDefault extends LitElement {
+  @state()
+  open: boolean = true;
+
+  private _openHandler() {
+    this.open = !this.open;
+  }
+
+  render() {
+    return html`
+      <div>
+        <cds-button @click=${this._openHandler}>Create partitions</cds-button>
+        <c4p-side-panel
+          include-overlay
+          ?open=${this.open}
+          size="md"
+          title="Create partitions"
+        >
```

Comment:
So something I noticed in this and in the stepped example, the side panel is controlled by the `open` state, but we aren't handling  the `open` state updates when a user is closing the side panel either by the keyboard, clicking outside of the sidepanel, or by clicking the "X" close button. What is happening currently appears to require two clicks to re-open the side panel since the first click after the user manually closes the component, the state value is still true, and the button is a toggle so it would on first click, set the state to false (nothing appears to happen for the user), and then the second click would then toggle back to true, showing the side panel... Adding in the event listener and forcing a state change would fix this:

```suggestion
        <c4p-side-panel
          include-overlay
          ?open=${this.open}
          size="md"
          title="Create partitions"
          @c4p-side-panel-closed=${this._handleClose}
        >
```
and before the render function, have something like this:
```JS
private _handleClose = () => {
  this.open = false;
};
```